### ISSUE TITLE
BlockLight is not being calculated

### ISSUE THREAD

**Issue by nic96** [lsehoon/bedrock2java#3]
bedrock edition does not store light values for blocks as far as I know, but java edition does so we'll have to write an algorithm to calculate light values for blocks based on how near a light emitting block is such as a torch. I may be wrong about how this works, but either way, the converted world has torches that don't seem to emit any light.

**Comment by lsehoon** [comment]
The task requires all map data to be transformed at same time. It may takes a long time to implement. Current logic is a simple implementation that drops shadows when sky light does not reach the block.

**Comment by nic96** [comment]
I started working on it, but I might give up. What I have now is code that updates the lighting for the chunk a torch is in and creates an array for blocks that need to be updated outside of the chunk so for this to fully work I would need a function to update existing *.mca region files based on the data in the array. It does update chunks beside the one the torch is in if that chunk gets processed after, but not if it happens before the one where the torch is in. Here's a screenshot of what I got:
![screenshot](https://i.imgur.com/VSszrtO.jpg)
